platform/android/export/export.cpp

```diff
@@ -2269,22 +2265,24 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 				keystore = p_preset->get("keystore/debug");
 				password = p_preset->get("keystore/debug_password");
 				user = p_preset->get("keystore/debug_user");
+			} else {
+				keystore = p_preset->get("keystore/release");
+				password = p_preset->get("keystore/release_password");
+				user = p_preset->get("keystore/release_user");
+			}
 
-				if (keystore.empty()) {
-					keystore = EditorSettings::get_singleton()->get("export/android/debug_keystore");
-					password = EditorSettings::get_singleton()->get("export/android/debug_keystore_pass");
-					user = EditorSettings::get_singleton()->get("export/android/debug_keystore_user");
-				}
+			if (keystore.empty()) {
+				// Fall back to the default editor keystore (not specific to the preset).
+				keystore = EditorSettings::get_singleton()->get("export/android/default_keystore");
+				password = EditorSettings::get_singleton()->get("export/android/default_keystore_pass");
+				user = EditorSettings::get_singleton()->get("export/android/default_keystore_user");
+			}
 
+			if (p_debug) {
 				if (ep.step("Signing debug APK...", 103)) {
```

Comment:
An explicit error message here when the keystore is not defined could help users, something like:
"No default keystore defined in Editor Settings and no debug keystore defined in Export settings, can't sign the debug APK."
(same thing for release)